### ISSUE TITLE
Implement system to ensure nupkg references are consistent with assembly references

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1133]
Inspired by #1126, which noticed that we'd released FakeItEasy 3.4.0,  whose assembly relied on Castle.Core 4.1.0, as a nupkg with a reference (for .NET Standard 1.6) to Castle.Core 4.0.0.

@holytshirt suggested using `dotnet pack` on the project instead of `nuget pack`, which is probably the ideal situation if it works, but @thomaslevesque thinks that this will create a separate symbols package instead of including the symbols in the nupkg. I'm also unsure that we'll get the right package versions, but I've not explored.

Another question I have about "dotnet pack" is how to create the analyzer package, including both C# and VB analyzers.

If it turns out that using "dotnet pack" isn't feasible, we could perhaps [expand the solution I used for SelfIntitializingFakes](https://github.com/blairconrad/SelfInitializingFakes/pull/2), where we read the referenced version from the csproj and use it to replace a property defined in the nuspec.
We'd need to be a little more sophisticated in FakeItEasy, though, since we have more dependencies and differing numbers of dependencies between frameworks (although so far we don't have different versions of dependencies for different frameworks).

**Comment by blairconrad** [comment]
> I'm also unsure that we'll get the right package versions, but I've not explored.

One can use a flag like `/p:Version=x.y.z` to specify the nupkg version.

But still no pdbs.

**Comment by thomaslevesque** [comment]
I just checked, `dotnet pack --include-symbols` does create a separate symbols package. I don't see any option to change this behavior.

> Another question I have about "dotnet pack" is how to create the analyzer package, including both C# and VB analyzers.

This doesn't seem to be possible 😞. But it's not a major problem in this case, since there are no dependencies.

**Comment by thomaslevesque** [comment]
> If it turns out that using "dotnet pack" isn't feasible, we could perhaps expand the solution I used for SelfIntitializingFakes, where we read the referenced version from the csproj and use it to replace a property defined in the nuspec.
> We'd need to be a little more sophisticated in FakeItEasy, though, since we have more dependencies and differing numbers of dependencies between frameworks (although so far we don't have different versions of dependencies for different frameworks).

It's an option, but it's not straightforward. There's no easy way to get the dependencies for each target framework, unless we want to parse the `Condition` attributes.

Anyway, the only platform that has NuGet dependencies is netstandard1.6, so maybe we don't need a very generic solution. I have something in mind, I'll try to send a PR soon.

**Comment by thomaslevesque** [comment]
FYI: https://github.com/NuGet/Home/issues/5440

**Comment by thomaslevesque** [comment]
We already have a working solution, but I just found a way to include the pdbs in the package with `dotnet pack`:

```xml
    <Content Include="$(OutputPath)/*/FakeItEasy.pdb">
      <Pack>true</Pack>
      <PackagePath>lib/%(RecursiveDir)FakeItEasy.pdb</PackagePath>
    </Content>
```

However `dotnet pack` doesn't seem to respect the `<PrivateAssets>All</PrivateAssets>` element, so it generates references to StyleCop...

**Comment by blairconrad** [comment]
Huh. 

1. cool
2. darn

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.0.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/4.0.0).

**Comment by thomaslevesque** [comment]
> I just checked, `dotnet pack --include-symbols` does create a separate symbols package. I don't see any option to change this behavior.

Related: https://github.com/NuGet/Home/issues/4142

I just realized something: the symbols package contains not only the PDB, but also the assembly! So I think we could just rename the .symbols.nupkg to .nupkg. Of course, we would just be trading one [hack](https://github.com/FakeItEasy/FakeItEasy/pull/1147/files#diff-ad5947a74d7c146b709f60c87bad9886R30) for another...
